lib/utils/utils.js

```diff
@@ -64,8 +66,46 @@ function setConsoleLogPrefix(prefix) {
     };
 }
 
+// This should be roughly similar to ARG_MAX, some systems is 131072 bytes
+// but we're going to be very conservative and cap the total size much lower than that to be safe.
+// POSIX suggestion is to go 2048 lower than the ARG_MAX value, as per https://www.in-ulm.de/~mascheck/various/argmax/
+const CMD_SIZE_LIMIT = 128 * 1024;
+const VAR_PATH = "./vars";
+
+/**
+ * Only allows a specific number of bytes for variable names and values.  Any variable exceeding that will be stored in a file
+ * and the variable value will contain the temp file location.
+ * If this substitution occurs then an additional variable called FILE_PARAMS will contain the list of substituted variable names.
+ * @param {Object} vars 
+ * @param {Number} sizeLimit Size limit to uphold (number of bytes)
+ * @returns Object of values from the provided one, modified as described.
+ */
+function limitVariableSizes(vars, sizeLimit = CMD_SIZE_LIMIT) {
+    const sizeLimitedVars = {};
+    for (const key in vars) {
+        const size = String(key).length + String(vars[key]).length;
```

Comment:
Use string interpolation? (`...string...` <= the backquotes)

Reply:
Both are valid stylistic choices, but I don't think it's worth re-triggering the build over it.

Reply:
Also in this case using interpolation to avoid the concatenation operator actually makes it longer overall.  I have to safely cast to String to get the length, so the interpolated version would look like this (except convert single quotes to backticks):
'${String(key).length}${String(vars[key]).length}' -- Not sure it looks much better this way.

Reply:
Since I want the length of a string, having javascript runtime parse a token pattern is less efficient than just calling the string function directly.  I don't think it sacrifices readability too much in this case.